using System;
using System.Reflection;
using log4net;

namespace WFT.PSService.Service
{
    //public class LoggedException : Exception
    //{
        //private readonly ILog _logger = LogManager.GetLogger( MethodBase.GetCurrentMethod( ).GetType( ).Name );

        ///// <summary>
        ///// Initializes a new instance of the <see cref="LoggedException"/> class.
        ///// </summary>
        //public LoggedException( )
        //{
        //    if( _logger.IsErrorEnabled )
        //        _logger.Error( "An unknown exception has occured during application processing." );
        //}

        ///// <summary>
        ///// Initializes a new instance of the <see cref="LoggedException"/> class.
        ///// </summary>
        ///// <param name="message">The message.</param>
        //public LoggedException( string message )
        //    : base( message )
        //{
        //    if( _logger.IsErrorEnabled )
        //        _logger.Error( message );
        //}

        ///// <summary>
        ///// Initializes a new instance of the <see cref="LoggedException"/> class.
        ///// </summary>
        ///// <param name="message">The message.</param>
        ///// <param name="innerException">The inner exception.</param>
        //public LoggedException( string message, Exception innerException )
        //    : base( message, innerException )
        //{
        //    if( _logger.IsErrorEnabled )
        //        _logger.Error( message, innerException );
        //}
    //}
}